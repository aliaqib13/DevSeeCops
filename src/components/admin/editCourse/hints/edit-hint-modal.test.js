import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import EditHintModal from './edit-hint-modal';

describe('EditHintModal', () => {
    const hint = { id: 5, course_id: 1, message: '<p>Test Hint</p>' };
    let component; let editHintMock; let toggleModalMock; let
        modal;

    beforeEach(() => {
        editHintMock = jest.fn(() => Promise.resolve(true));
        toggleModalMock = jest.fn();
        component = shallow(<EditHintModal hint={hint} />);
        component.setProps({ editHint: editHintMock, toggleModal: toggleModalMock });
        modal = component.find('Modal');
    });

    it('Should render successfully', () => {
        expect(modal.exists()).toBeTruthy();
        expect(toJson(component)).toMatchSnapshot();
    });

    it('Should render correct hint message in the input field', () => {
        expect(modal.props().children.props.children[1].props.data).toBe(component.instance().props.hint.message);
    });

    it('Should call editHint when modal is submitted', () => {
        modal.props().footer[1].props.onClick();
        expect(editHintMock).toBeCalledTimes(1);
    });

    it('Should call toggleModal when cancel button is clicked', () => {
        modal.props().footer[0].props.onClick();
        expect(toggleModalMock).toBeCalledTimes(1);
    });
});
