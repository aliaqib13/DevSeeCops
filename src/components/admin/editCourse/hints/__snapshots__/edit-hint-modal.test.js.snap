// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditHintModal Should render successfully 1`] = `
<Modal
  confirmLoading={false}
  footer={
    Array [
      <Button
        block={false}
        ghost={false}
        htmlType="button"
        loading={false}
        onClick={[MockFunction]}
      >
        Cancel
      </Button>,
      <Button
        block={false}
        ghost={false}
        htmlType="button"
        loading={false}
        onClick={[Function]}
        type="primary"
      >
        Edit
      </Button>,
    ]
  }
  maskTransitionName="fade"
  okType="primary"
  onCancel={[MockFunction]}
  title="Edit Hint"
  transitionName="zoom"
  visible={false}
  width={520}
>
  <div
    className="block-item"
  >
    <Title
      level={4}
    >
      Hint Message
    </Title>
    <CKEditor
      config={Object {}}
      data="<p>Test Hint</p>"
      id="hint_message"
      name="message"
      onBeforeLoad={[Function]}
      onChange={[Function]}
      readOnly={false}
      type="classic"
    />
  </div>
</Modal>
`;
